force-app/main/default/lwc/participantAdder/participantAdder.js

```diff
@@ -0,0 +1,137 @@
+/*
+ *
+ *  * Copyright (c) 2020, salesforce.com, inc.
+ *  * All rights reserved.
+ *  * SPDX-License-Identifier: BSD-3-Clause
+ *  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ *
+ */
+
+import { LightningElement, api, wire } from "lwc";
+import { format } from "c/util";
+import { getRecord } from "lightning/uiRecordApi";
+import { refreshApex } from "@salesforce/apex";
+import { ShowToastEvent } from "lightning/platformShowToastEvent";
+import getExistingParticipantContactIds from "@salesforce/apex/ServiceScheduleCreatorController.getExistingParticipantContactIds";
+import getServiceScheduleModel from "@salesforce/apex/ServiceScheduleCreatorController.getServiceScheduleModel";
+import addParticipants from "@salesforce/apex/ServiceScheduleCreatorController.addParticipants";
+
+import SUCCESS_LABEL from "@salesforce/label/c.Success";
+import SAVE_LABEL from "@salesforce/label/c.Save";
+import BACK_WARNING_LABEL from "@salesforce/label/c.Clicking_Back_Button_In_Modal_Warning";
+import CANCEL_LABEL from "@salesforce/label/c.Cancel";
+import LOADING_LABEL from "@salesforce/label/c.Loading";
+import SERVICE_FIELD from "@salesforce/schema/ServiceSchedule__c.Service__c";
+import SAVE_SUCCESS from "@salesforce/label/c.Add_Service_Participants_Success";
+
+export default class ParticipantAdder extends LightningElement {
+    @api recordId;
+    errorMessage;
+    isLoaded = false;
+    serviceScheduleModel;
+    existingParticipants;
+    labels = {
+        save: SAVE_LABEL,
+        success: SUCCESS_LABEL,
+        backWarning: BACK_WARNING_LABEL,
+        cancel: CANCEL_LABEL,
+        loading: LOADING_LABEL,
+        successMessage: SAVE_SUCCESS,
+    };
+
+    @wire(getRecord, {
+        recordId: "$recordId",
+        fields: [SERVICE_FIELD],
+    })
+    wiredSchedule(result) {
+        if (!(result.data || result.error)) {
+            return;
+        }
+        if (result.data) {
+            this.serviceId = result.data.fields[SERVICE_FIELD.fieldApiName].value;
+        }
+    }
+
+    @wire(getServiceScheduleModel, { recordTypeId: null })
+    wireServiceScheduleModel(result) {
+        if (!result) {
+            return;
+        }
+
+        if (result.data) {
+            this.serviceScheduleModel = JSON.parse(JSON.stringify(result.data));
+            this.extractLabels(this.serviceScheduleModel.labels.serviceParticipant);
```

Comment:
Do we need to even store the model after we extract the labels? I wonder if we should consider a future refactor so we can get the labels without getting the entire model.

Reply:
the child expects the model. i like the idea of a refactor later.